zpls_echograms/zpls_echogram.py

```diff
@@ -1,172 +1,374 @@
-import argparse, os, echopype
-from pathlib import Path
-from calendar import monthrange
-from datetime import date, timedelta
-from echopype.convert import Convert
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import argparse
+import cmocean
 import matplotlib.pyplot as plt
+import matplotlib.dates as mdates
 import numpy as np
+import os
+import warnings
+
+from calendar import monthrange
+from datetime import datetime, date, timedelta
+from echopype.convert import Convert
+from echopype.process import Process
 from pandas.plotting import register_matplotlib_converters
-import xarray as xr
+
 register_matplotlib_converters()
+subsite_config = {
+    'CE01ISSM': {
+        'long_name': 'Oregon Inshore Surface Mooring',
+        'tilt_correction': 15,
+        'colorbar_range': [-90, -50],
+        'vertical_range': [0, 25],
+        'deployed_depth': 25
+    },
+    'CE06ISSM': {
+        'long_name': 'Washington Inshore Surface Mooring',
+        'tilt_correction': 15,
+        'colorbar_range': [-90, -50],
+        'vertical_range': [0, 30],
+        'deployed_depth': 29
+    },
+    'CE07SHSM': {
+        'long_name': 'Washington Shelf Surface Mooring',
+        'tilt_correction': 15,
+        'colorbar_range': [-90, -50],
+        'vertical_range': [0, 87],
+        'deployed_depth': 87
+    },
+    'CE09OSSM': {
+        'long_name': 'Washington Offshore Surface Mooring',
+        'tilt_correction': 15,
+        'colorbar_range': [-90, -50],
+        'vertical_range': [0, 540],
+        'deployed_depth': 542
+    },
+    'CP04OSSM': {
+        'long_name': 'Offshore Surface Mooring',
+        'tilt_correction': 15,
+        'colorbar_range': [-150, 0],
+        'vertical_range': [0, 450],
+        'deployed_depth': 450
+    },
+    'CP03ISSM': {
+        'long_name': 'Inshore Surface Mooring',
+        'tilt_correction': 15,
+        'colorbar_range': [-150, 0],
+        'vertical_range': [0, 90],
+        'deployed_depth': 90
+    },
+    'CP01CNSM': {
+        'long_name': 'Central Surface Mooring',
+        'tilt_correction': 15,
+        'colorbar_range': [-150, 0],
+        'vertical_range': [0, 130],
+        'deployed_depth': 130
+    },
+    'GI02HYPM_Upper': {
+        'long_name': 'Apex Profiler Mooring, Upward Looking',
+        'tilt_correction': 15,
+        'colorbar_range': [-150, 0],
+        'vertical_range': [0, 150],
+        'deployed_depth': 150
+    },
+    'GI02HYPM_Lower': {
+        'long_name': 'Apex Profiler Mooring, Downward Looking',
+        'tilt_correction': 15,
+        'colorbar_range': [-150, 0],
+        'vertical_range': [150, 300],
```

Comment:
Colorbar range for Globals should be tightened up.  CG will need to determine what works.
If this is now vertical range, the downward-looking units should be 0 to something.... 500?